umbraco/Umbraco-CMS
# NetCore: Create new NuGet packages basis

Changes caused by the intentions of creating _UmbracoCMS.Core_, _UmbracoCMS.Web_ and _UmbracoCMS_ NuGet packages in accordance to .Net Core migrations.

> Note: 
**Umbraco.Examine.Lucene** is still framework, so further modifications need to be done in UmbracoCMS.Core.nuspec file later
**Umbraco.ModelsBuilder.Embedded** needs to be added later to UmbracoCMS.Web.nuspec file when the project is migrated

## Test
- In Powershell execute `$uenv=build/build.ps1 -get ` at the root folder to get the environment variable and download all the necessary tools
  - OBS: this needs to be done just the first time, after that, you can just use `$uenv=build/build.ps1 -get -loc` so you stop downloading over and over again and assume everything is ready (_build.temp_ folder will be deleted every time you run this command)
- Then `$uenv.CompileUmbraco()` will create a **bin** and **WebApp** folder into _build.temp_ with their contents
- Next command is `$uenv.PreparePackages()` which will create an additional **Configs** folder and will modify the contents of _WebApp_ dir 
- Finally, `$uenv.PackageNuGet()` will generate the 3 NuGet packages in **build.out** folder

---
❗ ❗ ❗ 
> This PR doesn't include 1873 other files that were modified due to copying and moving them around for creating the necessary setup for the successful creation of the NuGet packages! 

Some of the manual changes required are:
- make sure that you create XML files for the following projects (they are copied in the NuGet packages):
  - Umbraco.Configuration
  - Umbraco.Core
  - Umbraco.Examine.Lucene
  - Umbraco.Infrastrucure
  - Umbraco.PublishedCase.NuCache
  - Umbraco.Web.BackOffice
  - Umbraco.Web.Common
  - Umbraco.Web.Website
- Copy the files from _src\Umbraco.Web.UI\Umbraco_ to _src\Umbraco.Web.UI.NetCore\Umbraco_
  - if there are any problems with _cshtml_ files just delete their contents as dotnet core has precompiled views

---
Some of the changes in the build.ps1 script are rather hacky (_CompileUmbraco()_ ln170-ln189) but a lot of time was spent in investigating and wanted to put more efforts into other parts of the task.
Many lines are just commented out for any future use.
